cfpb/cfgov-refresh
# Filterable Lists: Prevent 500 error when filter year is < 1900

Bugfix: Previously, when a user entered a date earlier than 1/1/1900 in the Date Range fields on a Filterable List page, they would receive a 500 error. This happened because the `set_interpreted_date_values` method in this file uses `strftime` to format date input for display. `strftime` can only handle dates whose year is 1900 or later.

After this fix, when the user enters a date earlier than 1900, they see a validation error that says "Please enter a date later than 1/1/1900."

## Additions

- Added a validator on the `FilterableDateField` class that requires dates to be later than 1/1/1900

## Testing

1. Running this branch, visit any filterable list, e.g. http://localhost:8000/about-us/newsroom/
2. Open the filter controls, enter a date in the "from" and/or "to" field earlier than 1/1/1900.
3. Press "apply filters"
4. See that the page doesn't show the django error page. Instead, it shows a validation error specifying that dates must be later than 1/1/1900.

## Screenshots

![screen shot 2019-01-09 at 4 39 03 pm](https://user-images.githubusercontent.com/4295388/50930464-12532200-142e-11e9-8be7-dd0d636e43f0.png)

## Notes

- I chose 1/1/1900 as the cutoff date because that's the earliest date that our code (specifically the `strftime` method) can handle. I could choose a later date to be the cutoff if that would make more sense from a UX perspective.
- I made up the error message ("Please enter a date later than 1/1/1900."). What would be a better way to word it?

## Checklist
